Classes/Api/Controller/AbstractController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Oracle\Typo3Dam\Api\Controller;
+
+use GuzzleHttp\Client;
+
+abstract class AbstractController
+{
+    /**
+     * @var Client
+     */
+    protected $client;
+
+    /**
+     * @param Client $client
+     */
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @param string $id
+     *
+     * @return array
+     */
+    protected function getContentItem(string $id): array
+    {
+        $response = $this->client->request('GET', '/content/management/api/v1.1/items/' . $id);
```

Comment:
Would be nice to have a try and catch here, as it is an API request.

Reply:
Depends how thick we want to build this layer. If the remote side returns an error, it will result in a ClientException with the correct HTTP error code. We could convert that into custom exceptions here, but I'd rather leave it up to the repository layer to interpret the data, as the API layer never actually deals with the return schema.

E.g. 404s are handled in the repository:
https://github.com/pixelant/oracle_dam/blob/1257f5d6b3b1de8103e146419d6f8206deebc052/Classes/Domain/Repository/AssetRepository.php#L27-L37

Reply:
Alright fair point.